### PR TITLE
Added Giannini files

### PR DESCRIPTION
**Pull request by gcharles-chwy**
Please take a look.
UISlider and UITextField are included

### CODE REVIEW

File: BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/IBExamples/IBTextFieldViewController.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        myTextField.delegate = self
+    }
+    
+    @IBAction func typedInTextField(_ sender: Any) {
+        guard myTextField.text != nil else {return}
+        if myTextField.text!.count > 5 {
+            myTextField.text = "\(myTextField.text!.dropLast())"
+        }
```
Comment (ivancr) [inline BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/IBExamples/IBTextFieldViewController.swift]: A little cleaner and you don't force unwrap:
```
guard let text = myTextField.text, text.count > 5 else { return }

myTextField.text = "\(text.dropLast())"
```

File: BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/IBExamples/IBTextFieldViewController.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+}
+ 
+ extension IBTextFieldViewController: UITextFieldDelegate {
+    
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        guard myTextField.text != nil else {return false}
+        myTextFieldResultLabel.text = "You typed\n\(myTextField.text!)"
```
Comment (ivancr) [inline BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/IBExamples/IBTextFieldViewController.swift]: Same here:
```
guard let text = myTextField.text else { return false }
myTextFieldResultLabel.text = "You typed\n\(text)"
```

File: BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/ProgrammaticExamples/ProgrammaticSliderViewController.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        setupSlider()
+        setupBackground()
+        layoutConstraints()
+        }
+    
+    override func viewDidAppear(_ animated: Bool) {
+         self.playSong()
```
Comment (ivancr) [inline BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/ProgrammaticExamples/ProgrammaticSliderViewController.swift]:
```suggestion
         playSong()
```

**Review by ivancr — CHANGES_REQUESTED** [review]
Very small changes. Thanks so much for redoing it all again.

File: BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/IBExamples/IBTextFieldViewController.swift (hunk 2 above)
Comment (gcharles-chwy) [reply BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/IBExamples/IBTextFieldViewController.swift]: Refactored

File: BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/IBExamples/IBTextFieldViewController.swift (hunk 1 above)
Comment (gcharles-chwy) [reply BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/IBExamples/IBTextFieldViewController.swift]: Refactored.

File: BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/ProgrammaticExamples/ProgrammaticSliderViewController.swift (hunk 3 above)
Comment (gcharles-chwy) [reply BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Giannini/ProgrammaticExamples/ProgrammaticSliderViewController.swift]: Refactored

**Review by ivancr — APPROVED** [review]

**Comment by ivancr** [comment]
Nice job @GianniniCharles, you went above and beyond, thanks for that!

**Pull request merged**